### DISCUSSION TITLE
Multiple graphics in the same code block

### DISCUSSION THREAD

**Discussion by kevinpdavies** [interactive-data-language/vscode-idl#32] [Q&A] [answered]
I am trying our IDL Notebooks and I found that if I have two or more graphics outputs in the same code block only the final graphic appears. Is this the expected behaviour in IDL notebooks? A simple example that only produces a single output:

```
plot(/test)
surface(/test)
```

**Comment by znorman-harris** [comment]
You should have both outputs get embedded in the notebook (we actually have tests that check to make sure as part of our release process).

A couple of questions for you:

1. What version of IDL are you using?
2. What is the extension version?
3. What platform are you using (Windows, Mac, Arm Mac, Linux)?
4. If you restart VSCode, and re-run that cell, do you have any error messages pop up in the lower right of VSCode?

**Comment by kevinpdavies** [comment]
Hi Zachary,

1. My IDL version is 9.0
2. The extension is v4.5.0
3. I am using Mac M2 Pro
4. I restarted VSCode and re-ran the cell and it worked this time

Thank you Zachary - looks like I just needed that restart.

**Comment by kevinpdavies** [comment]
Hi Zachary,

So I think I was able to reproduce this by using this code block:
```
e = ENVI(/HEADLESS)
surface(/test)
plot(/test)
```
It seems that by creating the ENVI object, only the 2nd graphic appears. If I restart VSCode and comment out the first line, it works as expected.

**Comment by znorman-harris** [comment]
Ah, thanks! I can reproduce the error with that sample. Let me dig into it and see what's up.

**Comment by znorman-harris** [comment, marked as answer]
Alright, this has been fixed. Just published a new version of the extension (`4.5.1`) containing the fix. 

We also added some tests for this case to make sure it works as you expect in the future as well!

**Reply by kevinpdavies** [reply]
Great thank you Zachary for fixing this - it works for me now.
